cupy/core/_routines_manipulation.pyx

```diff
@@ -693,8 +705,8 @@ cdef _normalize_axis_tuple(axis, Py_ssize_t ndim,
 
 cdef ndarray _concatenate_single_kernel(
         list arrays, Py_ssize_t axis, tuple shape, dtype,
-        bint same_shape_and_contiguous):
-    cdef ndarray a, x, ret
+        bint same_shape_and_contiguous, out=None):
```

Comment:
As `_concatenate_single_kernel` is an internal function and called from only one place, I don't think `out` needs a default value.